### ISSUE TITLE
Shared directory

### ISSUE THREAD

**Issue by btopro** [psudug/nittany-vagrant#22]
we need to decide if and how we're going to tell people to manage this blackbox they are building (which it is currently).

SFTP vs shared directory vs ?

SFTP -- more performant / less messy
shared directory -- easier to understand, blurs the line between server and local development environment.

Personally, I'd lean towards shared directory and then give directions for other approaches if performance kills it for them. our default box is currently using a shared directory default of /vagrant in the box = the nittany-vagrant folder on the local file system.

Another potential option is to run rsync inside the VM and rsync files over to /vagrant/nittany. Then changes on local would be saved into /vagrant/nittany/whatever and then rsync inside the box would copy it over to /var/www/html/nittany/whatever. This might get around the performance issues as well as permission issues associated w/ shared directory.

Thoughts?

**Comment by heyMP** [comment]
Personally, I'm still in the camp of managing the files on the host machine, either shared directory or rsync.  Like Bryan said, it's just mentally easier to understand and  simpler than sftp.  @bradallenfisher probably knows best how much more performant rsync is over something like nfs.  

We'll definitely have to do benchmark comparisons between sharing/syncing and no-sharing in terms of the front-end stack.  There are so many files being changed and generated on the fly by node that it could dramatically decrease performance. If we are going to run the front-end stack in the virtual box then we might concider excluding these dynamically created files from being shared/synced.

**Comment by baf139** [comment]
benchmarking needed.

**Comment by btopro** [comment]
So the last option I put forth might be oddly the fastest. Let vm rsync to itself, that way drupal files actually being served in the box don't come from host machine but are copies of host machine

**Comment by btopro** [comment]
http://mitchellh.com/comparing-filesystem-performance-in-virtual-machines lots of numbers

**Comment by jpat14** [comment]
http://www.midwesternmac.com/blogs/jeff-geerling/nfs-rsync-and-shared-folder Some more interesting benchmark numbers. Definitely looks like rsync is the best option

**Comment by baf139** [comment]
agreed rsync is the best way for syncing. I will add the option in the Vagrantfile and we can sort out any permissions issues later.

**Comment by baf139** [comment]
passing to @jpat14 since he has some stuff in the works.
